packages/server/api/routes/variants.router.js

```diff
@@ -0,0 +1,39 @@
+const express = require('express');
+
+const router = express.Router({ mergeParams: true });
+
+// controllers
+const variantsController = require('../controllers/variants.controller');
+
+/**
+ * @swagger
+ * /variants/{productId}:
+ *  get:
+ *    tags:
+ *    - variants
+ *    summary: Get variants by productId
+ *    description:
+ *      Will return All variants with a matching productId.
+ *    produces: application/json
+ *    parameters:
+ *     - in: path
```

Comment:
I don't think this should be a path variable, I'd rather see it as a query parameter. 

Path segments are meant to express **resources**. So `/users` is the list of all user resources, `/users/{id}` gives you a specific User resource based on it's ID. 

So here, `/variants` gives you a list of all variants, and `/variants/{id}` should give me the variant with that ID but that's not what you mean here. 

You mean to **filter** the list of `variants` to only those that have the matching product ID. So I would say `/variants?product={ID}` instead.